src/mapml.js

```diff
@@ -961,12 +1057,56 @@ M.MapMLLayer = L.Layer.extend({
                     _leafletLayer: this,
                     crs: this.crs
                   }).addTo(map);
+                  this._setLayerElBounds();
                 }
               }, this);
         }
+        this._setLayerElBounds();
         this.setZIndex(this.options.zIndex);
         this.getPane().appendChild(this._container);
     },
+
+    //sets the <layer-> elements .bounds property 
+    _setLayerElBounds: function(){
+      var localBounds;
+      let layerTypes = ["_staticTileLayer","_imageLayer","_mapmlvectors","_templatedLayer"];
+      /* jshint ignore:start */
+      layerTypes.forEach((type) =>{
+        if(this[type]){
+          switch(type){
+            case "_templatedLayer":
+              for(let j =0;j<this[type]._templates.length;j++){
+                if(this[type]._templates[j].layer.layerBounds){
```

Comment:
If you drag [this layer](https://geogratis.gc.ca/mapml/en/cbmtile/fdi/) onto a map run by this PR, you'll get an undefined error: 
![image](https://user-images.githubusercontent.com/2437285/87477809-be169880-c5f6-11ea-915d-7e8f5429c401.png)

Reply:
I added a fix in commit ahmadayubi@3b8331020c9d9f9472afb61eb5188444964853fa for this error. It had to do with `<link rel="query">` and how those templates are handled. Should work now. Good catch.